### DISCUSSION TITLE
Is this how incremental builds works? Changing text in a single `<div>` is updating all pages

### DISCUSSION THREAD

**Discussion by sahandnayebaziz** [gatsbyjs/gatsby#35088] [Help] [answered]
This is picking up where [#12066](https://github.com/gatsbyjs/gatsby/issues/12066) left off:

When I change the text of a single `<div>` on my Contact page, (which is a single, isolated page that is not a subcomponent used by any other pages) all of the pages are being updated in the next build. I traced this to be due to a new diff appearing in the filename of `webpack-runtime-[HASH].js` which appears in every file. 

Is this expected behavior? I assumed that the incremental building process would mean that only `contact/index.html` would be updated, since that was the only page that seemed to have any code or output change.

**Comment by herecydev** [comment, marked as answer]
Yes that's correct. A code change (no matter how simple) is not "traced" through your application, so whilst you only change a file that's used by a single page. The webpack hash will change and a whole rebuild is triggered.

Incremental builds would have helped if the **data** for your page changes; the HTML file and associated json would only need to change

**Reply by sahandnayebaziz** [reply]
That is great to know. Thank you for explaining it so clearly. You've helped me understand Gatsby quite a bit better!

**Reply by KyleAMathews** [reply]
Basically we're as incremental as possible. If code changes we do have to rebuild a lot but there's still savings vs a completely cold build with no cache. Sometimes considerable still.
